src/software/ai/move_rule/move_rule_manager_test.cpp

```diff
@@ -0,0 +1,162 @@
+
+    game_state.updateRefboxGameState(RefboxGameState::NORMAL_START);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::DIRECT_FREE_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::INDIRECT_FREE_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::TIMEOUT_US);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::TIMEOUT_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::GOAL_US);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::GOAL_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::BALL_PLACEMENT_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+}
+
+TEST(MoveRuleTest, test_avoid_areas_from_game_start_or_us_states)
```

Comment:
we should remove all references to `avoid_areas`

Reply:
Whoops